### PR TITLE
src: remove unused headers-exported functions

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
None : generic cleaning
https://redmine.openinfosecfoundation.org/issues/4083

Describe changes:
- remove functions defined in a header and never used (debug, api-completeness, and others)
- remove double definition of IPPairLock

Python helper script to do this in some previous PR

#10684 with needed rebase

### CODE REVIEW

File: src/detect-engine.c [hunk 1]
```diff
@@ -1250,14 +1203,6 @@ void DetectEngineBufferTypeSupportsFrames(DetectEngineCtx *de_ctx, const char *n
     SCLogDebug("%p %s -- %d supports frame inspection", exists, name, exists->id);
 }
 
-void DetectEngineBufferTypeSupportsPacket(DetectEngineCtx *de_ctx, const char *name)
```
Comment (victorjulien) [inline src/detect-engine.c:1253]: would like to keep this for completeness of the api

File: src/suricata.c [hunk 2]
```diff
@@ -237,12 +237,6 @@ int EngineModeIsIPS(void)
     return (g_engine_mode == ENGINE_MODE_IPS);
 }
 
-int EngineModeIsIDS(void)
```
Comment (victorjulien) [inline src/suricata.c:240]: would like to keep for completeness of the api

File: src/util-bloomfilter.h [hunk 3]
```diff
@@ -35,10 +35,7 @@ typedef struct BloomFilter_ {
 /* prototypes */
 BloomFilter *BloomFilterInit(uint32_t, uint8_t, uint32_t (*Hash)(const void *, uint16_t, uint8_t, uint32_t));
```
Comment (victorjulien) [inline src/util-bloomfilter.h:36]: the whole bloom filter code is unused I think, so might want to just remove all of it (separate commit)

File: src/util-byte.c [hunk 4]
```diff
@@ -644,31 +644,6 @@ int StringParseInt32(int32_t *res, int base, size_t len, const char *str)
     return ret;
 }
 
-int StringParseInt16(int16_t *res, int base, size_t len, const char *str)
```
Comment (victorjulien) [inline src/util-byte.c:647]: keep for completeness

File: src/util-lua-common.c [hunk 5]
```diff
@@ -89,19 +89,6 @@ void LuaPushTableKeyValueInt(lua_State *luastate, const char *key, int value)
  *
  *  If value is NULL, string "(null")" will be put on the stack.
  */
-void LuaPushTableKeyValueString(lua_State *luastate, const char *key, const char *value)
```
Comment (victorjulien) [inline src/util-lua-common.c:92]: keep for completeness

File: src/util-misc.c [hunk 6]
```diff
@@ -144,23 +144,6 @@ static int ParseSizeString(const char *size, double *res)
     return retval;
 }
 
-int ParseSizeStringU8(const char *size, uint8_t *res)
```
Comment (victorjulien) [inline src/util-misc.c:147]: keep for completeness

File: src/util-path.c [hunk 7]
```diff
@@ -117,42 +117,6 @@ char *PathMergeAlloc(const char *const dir, const char *const fname)
     return ret;
 }
 
-/**
```
Comment (victorjulien) [inline src/util-path.c:120]: think I'd like to keep this too

File: src/util-path.c [hunk 8]
```diff
@@ -244,37 +208,6 @@ bool SCIsRegularDirectory(const struct dirent *const dir_entry)
 #endif
     return false;
 }
-/**
```
Comment (victorjulien) [inline src/util-path.c:247]: lets keep these

File: src/util-radix-tree.c [hunk 9]
```diff
@@ -1355,21 +1355,6 @@ static void SCRadixRemoveKey(uint8_t *key_stream, uint16_t key_bitlen,
     return;
 }
 
-/**
```
Comment (victorjulien) [inline src/util-radix-tree.c:1358]: separately, it would be good to remove the unused "Generic" support from the radix code

**Review by victorjulien — CHANGES_REQUESTED** [review]
mostly ok, but there are number of things I'd like to keep, see inline

File: src/conf.c [hunk 10]
```diff
@@ -583,25 +583,6 @@ int ConfValIsFalse(const char *val)
  * \retval 1 will be returned if the name is found and was properly
  * converted to a double, otherwise 0 will be returned.
  */
-int ConfGetDouble(const char *name, double *val)
```
[the remark this answers is not in the document]
Comment (catenacyber) [reply src/conf.c:586]: ok

File: src/detect-engine.c (hunk 1 above)
Comment (catenacyber) [reply src/detect-engine.c:1253]: ok

File: src/suricata.c (hunk 2 above)
Comment (catenacyber) [reply src/suricata.c:240]: ok

File: src/util-byte.c (hunk 4 above)
Comment (catenacyber) [reply src/util-byte.c:647]: ok

File: src/util-lua-common.c (hunk 5 above)
Comment (catenacyber) [reply src/util-lua-common.c:92]: ok

File: src/util-path.c (hunk 8 above)
Comment (catenacyber) [reply src/util-path.c:247]: ok

File: src/util-misc.c (hunk 6 above)
Comment (catenacyber) [reply src/util-misc.c:147]: ok

File: src/util-radix-tree.c (hunk 9 above)
Comment (catenacyber) [reply src/util-radix-tree.c:1358]: > unused "Generic" support from the radix code

What is this ?

**Comment by catenacyber** [comment]
Replaced by https://github.com/OISF/suricata/pull/10827

File: src/util-radix-tree.c (hunk 9 above)
Comment (victorjulien) [reply src/util-radix-tree.c:1358]: the radix code can store 3 types of data: ip4, ip6 and "generic". But generic isn't used.

Comment (catenacyber) [reply src/util-radix-tree.c:1358]: Not sure I see it : for instance
`SCRadixAddKey` is only called by IPv4 and IPv6 functions

**Pull request closed without merging**